[16.0][IMP] stock_quantity_history_location: invisible location except from stock.quant view

AungKoKoLin1997 opened the pull request:
This PR makes the location field invisible when the model is not stock.quant (locations view), because the location context is not used in other models (e.g., `stock.valuation.layer`) and choosing location will not effect the functionality. So, displaying this field in the pop-up of other models (e.g., Stock Valuation Layer) may confuse users.

@qrtl QT4756

AungKoKoLin1997 committed:
[IMP] stock_quantity_history_location: invisible location except from stock.quant view

yostashiro reviewed (changes requested):
I think `include_child_locations` should be removed. See https://github.com/odoo/odoo/pull/78078.

Can you please also update the README accordingly, as the usage seems outdated.

AungKoKoLin1997 committed:
[IMP] stock_quantity_history_location: remove deadcode

Odoo core behavior has changed, so this module’s redundant code can be removed:
- Drop the include_child_locations flag: child locations are now included by default (see https://github.com/odoo/odoo/pull/78078).
- Remove the company_owned context: it was removed upstream in commit https://github.com/odoo/odoo/commit/6fd05759e7f77f010dbd3fb8a9960128f4ee9fa1#diff-2f7d1a089caccb39f6dcaf4afe19af96ebf382886e382b22e1e956ba94a927c9.

AungKoKoLin1997 committed:
[IMP] stock_quantity_history_location: improve readme

AungKoKoLin1997 commented:
> I think `include_child_locations` should be removed. See [odoo/odoo#78078](https://github.com/odoo/odoo/pull/78078).
> 
> Can you please also update the README accordingly, as the usage seems outdated.

@yostashiro Done!

yostashiro commented on stock_quantity_history_location/tests/test_stock_quantity_history_location.py:
```diff
@@ -31,35 +31,43 @@ def setUpClass(cls):
+        action = self.get_stock_quantity_history_action(
+            self.child_test_stock_loc, current_date
+        )
+        to_date_in_context = action["context"].get("to_date")
+        self.assertIsNotNone(to_date_in_context)
+        self.assertEqual(to_date_in_context, current_date)
```
Non-blocking. I think this should be sufficient.
```suggestion
        self.assertEqual(action["context"].get("to_date"), current_date)
```

yostashiro reviewed (approved):
LGTM.

nobuQuartile reviewed (approved):
Functional review
LGTM

AungKoKoLin1997 commented:
@oca/stock-logistics-reporting-maintainers
Can we get this PR merged?

rolandojduartem commented on stock_quantity_history_location/wizards/stock_quantity_history.xml:
```diff
@@ -7,11 +7,10 @@
-                    attrs="{'invisible': [('location_id', '=', False)]}"
+                    name="location_id"
                     groups="stock.group_stock_multi_locations"
+                    invisible="context.get('active_model') != 'stock.quant'"
```
does this work on Odoo 16.0? I think it should be attrs

AungKoKoLin1997 replied:
@rolandojduartem 
It works on odoo 16.0. You can see how odoo standard field `inventory_datetime` displays in view depends on model.
https://github.com/odoo/odoo/blob/1adee2b7452c072bcce51cd75641a9e4c2b7f219/addons/stock_account/wizard/stock_quantity_history.xml#L9-L19.

kanda999 reviewed (approved):
Functional review: it works in prod server

rousseldenis commented:
/ocabot merge minor
